### PR TITLE
(PDB-3808) Remove node_state 'any' nodes during query engine processing

### PR DESCRIPTION
**Pull request by ajroetker**

### CODE REVIEW

File: src/puppetlabs/puppetdb/query_eng/engine.clj [hunk 1]
```diff
@@ -1526,6 +1526,19 @@
...
+(defn remove-elided-nodes
+  "This step removes elided nodes (marked with ::elide by expand-user-query) from
+  `and` and `or` clauses."
+  [node]
+  (cm/match [node]
+            [["and" & clauses]]
+            (into ["and"] (remove #(= ::elide %) clauses))
+
+            [["or" & clauses]]
+            (into ["or"] (remove #(= ::elide %) clauses))
```
Comment (mullr) [inline src/puppetlabs/puppetdb/query_eng/engine.clj]: Is there any way this can happen outside of an "and" or "or"?

Comment (ajroetker) [reply src/puppetlabs/puppetdb/query_eng/engine.clj]: what other case are you thinking of?

Comment (mullr) [reply src/puppetlabs/puppetdb/query_eng/engine.clj]: maybe if you just had node_state== any by itself? it seems like that would get ::elide, but not hit iether of these cases.

**Comment by ajroetker** [comment]
@mullr Cleaned this up by simplifying to just remove elided values for all vector nodes!

**Comment by rbrw** [comment]
Russ says go, and AJ says go, so GO!

**Pull request merged**